src/Disks/TemporaryFileOnDisk.h

```diff
@@ -23,7 +23,9 @@ class TemporaryFileOnDisk
     ~TemporaryFileOnDisk();
 
     DiskPtr getDisk() const { return disk; }
+    /// Return absolute path (disk + relative_path)
     String getPath() const;
+    const String & getRelativePath() const { return relative_path; }
```

Comment:
Nice catch!

Do we actually need absolute paths for anything?
I mean what if we change `getPath()` so it returns the relative path instead of adding the new function `getRelativePath()`?